speakers.py

```diff
@@ -0,0 +1,321 @@
+        color: #000;
+        font-size: 20px;
+        transition: color 0.3s;
+    }}
+    .social-links a:hover {{
+        color: #007bff; /* Change to your desired hover color */
+    }}
+    .biography {{
+        max-height: 100px;
+        overflow: hidden;
+        position: absolute;
+        bottom: 0;
+        left: 0;
+        width: 100%;
+        padding: 10px;
+        background: rgba(0, 0, 0, 0.8);
+        color: #fff;
+        transition: max-height 0.3s, opacity 0.3s;
+        box-sizing: border-box;
+    }}
+    .biography:hover {{
+        max-height: 200px; /* Adjust based on how much you want to show */
+        overflow-y: auto;
+    }}
+</style>
+
+<div class="container">{speakers}</div>
+
+"""
+SPEAKER_CARD = """
+    <div class="speaker-card">
+        <img src="{avatar}" class="avatar">
+        <div class="info">
+            <div class="name">{name}</div>
+            <div class="role">{role}</div>
+            <div class="social-links">{social_links}</div>
+        </div>
+        <div class="biography">
+            <p>{bio}</p>
+        </div>
+    </div>
+"""
+
+
+def fix_twitter(url):
+    new_url = url.replace('http://', 'https://').lower()
+    if 'https' in new_url:
+        return new_url
+    if new_url:
+        return f"https://twitter.com/{new_url}"
+    return ''
+
+
+def fix_github(url):
+    new_url = url.replace('http://', 'https://').lower()
+    if 'https' in new_url:
+        return new_url
+    if new_url:
+        return f"https://github.com/{new_url}"
+    return ''
+
+
+def fix_linkedin(url):
+    new_url = url.replace('http://', 'https://').lower()
+    if 'https' in new_url:
+        return new_url
+    if new_url:
+        return f"https://linkedin.com/in/{new_url}"
+    return ''
+
+def fix_facebook(url):
+    return url
+
+
+SOCIAL_ICONS = {
+    'twitter': ('fab fa-twitter', fix_twitter),
+    'github': ('fab fa-github', fix_github),
+    'linkedin': ('fab fa-linkedin', fix_linkedin),
+    'facebook': ('fab fa-facebook', fix_facebook),
+}
+
+
+def generate_speakers_page(speakers):
+
+    speakers_list = []
+    for speaker in speakers:
+        avatar = speaker['avatar'] if speaker['avatar'].strip() else './theme/img/PyConIL.png'
+        role = speaker['answers']['role'] if 'role' in speaker['answers'] else ''
+        company = speaker['answers']['company'] if 'company' in speaker['answers'] else ''
+
+        social_links = []
+        for social, value in speaker['answers'].items():
+            if social in SOCIAL_ICONS:
+                stype = SOCIAL_ICONS[social][0]
+                url = SOCIAL_ICONS[social][1](value)
+                social_links.append(f'<a href="{url}" target="_blank"><i class="{stype}"></i></a>')
+
+        social_links_html = '\n'.join(social_links)
+        speakers_list.append(SPEAKER_CARD.format(
+            avatar=avatar,
+            name=speaker['name'].title(),
+            role=f"{role} @ {company}",
+            bio=speaker['biography'],
```

Comment:
```suggestion
            bio=speaker['biography'] or '',
```

to prevent this: 
![image](https://github.com/user-attachments/assets/73fef9dd-6724-42e6-bbf7-d657bb17d8cd)